### PR TITLE
ddl:support alter table comment

### PR DESCRIPTION
**Pull request by ciscoxll**
Added a command support alter table comment respond issues #4990 `alter table mytable comment 'this is table comment';`.

### CODE REVIEW

**Comment by ciscoxll** [comment]
@zimulala @winkyao PTAL

**Comment by winkyao** [comment]
What is the related github issue?

File: model/ddl.go [hunk 1]
```diff
@@ -45,6 +45,7 @@ const (
 	ActionRenameTable     ActionType = 14
 	ActionSetDefaultValue ActionType = 15
 	ActionShardRowID      ActionType = 16
+	ActionSetTableComment ActionType = 17
```
Comment (winkyao) [inline model/ddl.go]: s/ActionSetTableComment/ActionModifyTableComment/

File: ddl/table.go [hunk 2]
```diff
@@ -335,6 +335,30 @@ func (d *ddl) onRenameTable(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 	return ver, nil
 }
 
+func (d *ddl) onSetTableComment(t *meta.Meta, job *model.Job) (ver int64, _ error) {
```
Comment (winkyao) [inline ddl/table.go]: s/onSetTableComment/onModifyTableComment/

File: ddl/ddl_api.go [hunk 3]
```diff
@@ -1436,6 +1439,31 @@ func (d *ddl) AlterColumn(ctx sessionctx.Context, ident ast.Ident, spec *ast.Alt
 	return errors.Trace(err)
 }
 
+func (d *ddl) TableComment(ctx sessionctx.Context, ident ast.Ident, spec *ast.AlterTableSpec) error {
```
Comment (winkyao) [inline ddl/ddl_api.go]: s/TableComment/AlterTableComment/

Comment (winkyao) [reply ddl/ddl_api.go]: Please add comments before exported method

File: ddl/ddl_api.go [hunk 4]
```diff
@@ -1436,6 +1439,31 @@ func (d *ddl) AlterColumn(ctx sessionctx.Context, ident ast.Ident, spec *ast.Alt
...
+	if err != nil {
+		return errors.Trace(infoschema.ErrTableNotExists.GenByArgs(ident.Schema, ident.Name))
+	}
+
+	job := &model.Job{
+		SchemaID:   schema.ID,
+		TableID:    tb.Meta().ID,
+		Type:       model.ActionSetTableComment,
+		BinlogInfo: &model.HistoryInfo{},
+		Args:       []interface{}{schema.ID, spec.Name, spec.Comment},
```
Comment (winkyao) [inline ddl/ddl_api.go]: Arg `schema.ID` is unnecessary, you can use job.SchemaID

**Comment by ciscoxll** [comment]
@zimulala @winkyao PTAL

File: ddl/table.go [hunk 5]
```diff
@@ -335,6 +335,29 @@ func (d *ddl) onRenameTable(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 	return ver, nil
 }
 
+func (d *ddl) onModifyTableComment(t *meta.Meta, job *model.Job) (ver int64, _ error) {
+	var tableName string
+	var comment string
+	if err := job.DecodeArgs(&tableName, &comment); err != nil {
```
Comment (winkyao) [inline ddl/table.go]: tableName is useless?

**Comment by ciscoxll** [comment]
@zimulala @winkyao PTAL

**Comment by ciscoxll** [comment]
@zimulala  PTAL

File: ddl/table.go [hunk 6]
```diff
@@ -335,6 +335,28 @@ func (d *ddl) onRenameTable(t *meta.Meta, job *model.Job) (ver int64, _ error) {
...
+
+	tblInfo, err := getTableInfo(t, job, job.SchemaID)
+	if err != nil {
+		return ver, errors.Trace(err)
+	}
+
+	tblInfo.Comment = comment
+	ver, err = updateVersionAndTableInfo(t, job, tblInfo, true)
+	if err != nil {
+
```
Comment (zimulala) [inline ddl/table.go]: Please remove this blank line.

File: ddl/ddl_api.go [hunk 7]
```diff
@@ -1436,6 +1439,32 @@ func (d *ddl) AlterColumn(ctx sessionctx.Context, ident ast.Ident, spec *ast.Alt
...
+// AlterTableComment updates the table comment information.
+func (d *ddl) AlterTableComment(ctx sessionctx.Context, ident ast.Ident, spec *ast.AlterTableSpec) error {
+	is := d.infoHandle.Get()
+	schema, ok := is.SchemaByName(ident.Schema)
+	if !ok {
+		return infoschema.ErrDatabaseNotExists.GenByArgs(ident.Schema)
+	}
+
+	tb, err := is.TableByName(ident.Schema, ident.Name)
+	if err != nil {
```
Comment (zimulala) [inline ddl/ddl_api.go]: Please add some tests like `TestMySQLErrorCode` in ddl_db_test.go

**Comment by ciscoxll** [comment]
@zimulala PTAL

**Comment by shenli** [comment]
@coocood @zimulala PTAL

**Pull request merged**